### PR TITLE
fix issue 1166: treat all zk child path url as new child

### LINKED ISSUE #1166: when invoker refer check fail,  service will be added in cache invokers fail
**What happened**:
接到zk新服务上线通知缓存invoker的时候会进行服务的refer检查，如果当时服务有问题，或者还没完全启动，缓存就加不进去 ，就会导致这个provider服务一直发现不了。 
image

image

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by AlexStocks**
**What this PR does**:

treat all zk child path url as new child

**Which issue(s) this PR fixes**:

Fixes #1166

### CODE REVIEW

**Review by Mulavar — APPROVED** [review]

File: remoting/zookeeper/listener.go [hunk 1]
```diff
@@ -323,12 +319,24 @@ func (l *ZkEventListener) listenDirEvent(conf *common.URL, zkPath string, listen
 			}
 		}
 		// Periodically update provider information
-		ticker := time.NewTicker(ttl)
+		tickerTTL := ttl
```
Comment (Mulavar) [inline remoting/zookeeper/listener.go:322]: if ttl is less than 20e9, the timer may trigger too frequentyl?

**Review by zhaoyunxing92 — APPROVED** [review]
approved these changes

File: remoting/zookeeper/listener.go (hunk 1 above)
Comment (AlexStocks) [reply remoting/zookeeper/listener.go:322]: 用户想咋用我们不管，设置太长，我们在客户端启动时做个弹性。

**Review by zouyx — APPROVED** [review]

**Review by xavier-niu — APPROVED** [review]

File: remoting/zookeeper/listener.go [hunk 2]
```diff
@@ -160,10 +160,6 @@ func (l *ZkEventListener) handleZkNodeEvent(zkPath string, children []string, li
 		newNode string
 	)
 	for _, n := range newChildren {
-		if contains(children, n) {
```
[the remark this answers is not in the document]
Comment (wenxuwan) [reply remoting/zookeeper/listener.go:163]: 这个地方去掉会发生重复监听的。不应该去掉

**Pull request merged**